crates/bevy_animation/src/lib.rs

```diff
@@ -691,217 +740,308 @@ where
         + tangent_in_end * step_duration * (lerp.powi(3) - lerp.powi(2))
 }
 
-#[allow(clippy::too_many_arguments)]
-fn apply_animation(
-    weight: f32,
+fn update_transitions(player: &mut AnimationPlayer, time: &Time) {
+    player.transitions.retain_mut(|animation| {
+        animation.current_weight -= animation.weight_decline_per_sec * time.delta_seconds();
+        animation.current_weight > 0.0
+    });
+}
+
+/// Adds animation support to an app
+#[derive(Default)]
+pub struct AnimationPlugin;
+
+impl Plugin for AnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_asset::<AnimationClip>()
+            .register_asset_reflect::<AnimationClip>()
+            .register_type::<AnimationPlayer>()
+            .add_systems(Update, update_animation_targets)
```

Comment:
If `Update` is where we expect users to do most of their work, there's a good chance they reparent entities there, no? Shouldn't we settle on a high level "Animations run in `PostUpdate`", and then have some system sets inside `PostUpdate` to order the various animation jobs? I don't like for built-in core systems like this to do anything in `Update`, this increases the likelyhood of system ordering bugs and makes it more difficult for new users. I know advanced users won't have an issue because they will order their system(s) relative to this one, but that's not a trivial thing to do, and that's also annoying to maintain at the minute between Bevy releases.

Reply:
Yeah, I didn't like doing things in `Update`. I only did it because we need to flush the `Commands` in between the two phases. I can look to see if there's another place to put it in `PostUpdate`.

Reply:
No longer relevant.